IdentityCore/tests/MSIDMacKeychainTokenCacheTests.m

```diff
@@ -0,0 +1,434 @@
+    // have the same primary key values
+    MSIDAccountCacheItem *deletedAccountA = [_dataSource accountWithKey:key1
+                                                             serializer:_serializer
+                                                                context:nil
+                                                                  error:&error];
+    XCTAssertNil(deletedAccountA);
+}
+
+- (void)testRemoveItemsWithAccountKey_whenKeyIsValid_shouldRemoveItem
+{
+    NSError *error;
+    BOOL result = [_dataSource saveAccount:_testAccount key:_testAccountKey serializer:_serializer context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    MSIDAccountCacheItem *account2 = [_dataSource accountWithKey:_testAccountKey
+                                                      serializer:_serializer
+                                                         context:nil
+                                                           error:&error];
+    XCTAssertNil(error);
+    XCTAssertEqualObjects(_testAccount, account2);
+
+    result = [_dataSource removeItemsWithAccountKey:_testAccountKey context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    account2 = [_dataSource accountWithKey:_testAccountKey serializer:_serializer context:nil error:&error];
+    XCTAssertNil(error);
+    XCTAssertNil(account2);
+}
+
+- (void)testAccountsWithKey_whenAccountMissing_shouldNotReturnError
+{
+    NSError *error;
+    NSString* accountId = @"AnotherTestAccountId";
+    MSIDDefaultAccountCacheKey *key = [[MSIDDefaultAccountCacheKey alloc] initWithHomeAccountId:accountId
+                                                                                    environment:_testAccount.environment
+                                                                                          realm:_testAccount.realm
+                                                                                           type:_testAccount.accountType];
+    _testAccountKey.username = _testAccount.username;
+
+    // make sure it's not there
+    BOOL result = [_dataSource removeItemsWithAccountKey:key context:nil error:&error];
+    XCTAssertTrue(result);
+    XCTAssertNil(error);
+
+    // verify we can't retrieve it
+    MSIDAccountCacheItem *account2 = [_dataSource accountWithKey:key serializer:_serializer context:nil error:&error];
+    XCTAssertNil(error); // "not found" isn't an error
+    XCTAssertNil(account2);
+}
+
+- (void)testAccountsWithQuery_whenMultipleAccountsPresent_shouldReturnExpectedAccounts
+{
+    [self multiAccountTestSetup];
+
+    // Verify reading multiple accounts returns the expected accounts:
+    NSError *error;
+    NSArray<MSIDAccountCacheItem *> *accountList;
+    accountList = [_cache getAccountsWithQuery:_queryAll context:nil error:&error];
+    XCTAssertNil(error);
+    XCTAssertTrue(accountList.count == 3);
+    for (MSIDAccountCacheItem *account in accountList)
```

Comment:
how do we ensure all 3 get returned?

Reply:
I improved the test check to verify this